### PR TITLE
Update yarn rules

### PR DESCRIPTION
**Pull request by jakechorleytm**

### CODE REVIEW

File: js/yarn.build_defs [hunk 1]
```diff
@@ -44,17 +44,22 @@ def yarn_module(name, version, package_name=None, resolved_by:str='', out=None,
...
-def _yarn_install(third_party_location, production):
-    return [
-        "cd $PKG_DIR",
+def _yarn_install(third_party_location, production, environment_vars):
+    cmd = ["cd $PKG_DIR"]
+
+    for var in environment_vars.keys():
+        value = environment_vars[var]
+        cmd.append(f"export {var}={value}")
+
```
Comment (jakechorleytm) [inline js/yarn.build_defs:53]: Allows you to set env vars before the install. For instance telling node sass where to find the post install binary

File: js/yarn_deps/index.js [hunk 2]
```diff
@@ -1,164 +1,176 @@
-const fs = require('fs');
```
Comment (jakechorleytm) [inline js/yarn_deps/index.js:1]: Prettier attacked this file when I made some small changes. Have pointed them out

File: js/yarn_deps/index.js [hunk 3]
```diff
@@ -1,164 +1,176 @@
...
+    console.log(`
 yarn_module(
-    name = '${name}',
-    out = '${out}',
-    resolved_by = '${resolvedBy}',
-    version = '${version}',
+    name = "${name}",
+    out = "${out}",
+    resolved_by = "${resolvedBy}",
+    version = "${version}",
```
Comment (jakechorleytm) [inline js/yarn_deps/index.js:53]: Changed ' to "

File: js/yarn_deps/index.js [hunk 4]
```diff
@@ -1,164 +1,176 @@
...
+  name = name.replace("/", "--");
+
+  // If there is some version for this dependency greater than this version, add the version number to the rule name
+  // assuming this will only be depended on by other module rules
+  if (
+    versions.some(function (ver) {
+      return semver.gt(ver, version);
+    })
+  ) {
+    return `${name}@${version}`;
```
Comment (jakechorleytm) [inline js/yarn_deps/index.js:73]: This now prefixes version with @.

File: js/yarn_deps/index.js [hunk 5]
```diff
@@ -1,164 +1,176 @@
...
-    // assuming this will only be depended on by other module rules
-    if(versions.some(function (ver) {
-        return semver.gt(ver, version)
-    })) {
-        return `${name}-${version}`
-    }
-    // Otherwise we are the latest version, use the raw name so others can depend on us easily
-    return name
+  let versions = packageVersions[name];
+  name = name.replace("/", "--");
```
Comment (jakechorleytm) [inline js/yarn_deps/index.js:64]: Now replacing / with -- to avoid possible naming clashes

**Comment by Tatskaari** [comment]
Thanks Jake! I'll have to figure out why the ci stuff is failing but I'm sure it's nothing to do with your changes.

**Pull request merged**